test/attachArtifact.test.js

```diff
@@ -0,0 +1,193 @@
+'use strict'
+
+const tap = require('tap')
+
+const DEFAULT_INPUT_DATA = {
+  artifactPath: 'dist',
+  artifactFilename: 'filename.zip',
+  artifactLabel: 'label',
+  releaseId: '1',
+  token: 'token',
+}
+
+tap.test(
+  'attachArtifact does not throw errors with proper inputs',
+  async () => {
+    const attachArtifactModule = tap.mock('../src/utils/attachArtifact.js', {
+      '../src/utils/archiver.js': {
+        archiveItem: () => null,
+      },
+      'fs/promises': {
+        stat: () => 100,
+        lstat: () => ({ isDirectory: () => true }),
+        readFile: async () => Buffer.from([0x62, 0x75, 0x66, 0x66, 0x65, 0x72]),
+      },
+      '@actions/github': {
+        context: {
+          repo: {
+            repo: 'repo',
+            owner: 'owner',
+          },
+        },
+        getOctokit: () => ({
+          rest: {
+            repos: {
+              uploadReleaseAsset: async () => ({ status: 201, data: {} }),
+            },
+          },
+          request: async () => ({
+            data: { browser_download_url: 'https://example.com' },
+          }),
+        }),
+      },
+    })
+
+    const { artifactPath, artifactFilename, artifactLabel, releaseId, token } =
+      DEFAULT_INPUT_DATA
+    tap.doesNotThrow(async () => {
```

Comment:
there are better methods to check promise resolution/rejection

Reply:
could you please name a few? I found only this one in the node-tap documentation. Thank you

Reply:
https://node-tap.org/docs/api/asserts/#trejectspromise--fn-expectederror-message-extra